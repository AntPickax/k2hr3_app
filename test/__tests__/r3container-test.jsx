/*
 *
 * K2HR3 Web Application
 *
 * Copyright 2017 Yahoo! Japan Corporation.
 *
 * K2HR3 is K2hdkc based Resource and Roles and policy Rules, gathers
 * common management information for the cloud.
 * K2HR3 can dynamically manage information as "who", "what", "operate".
 * These are stored as roles, resources, policies in K2hdkc, and the
 * client system can dynamically read and modify these information.
 *
 * For the full copyright and license information, please view
 * the license file that was distributed with this source code.
 *
 * AUTHOR:   Takeshi Nakatani
 * CREATE:   Thu Feb 15 2018
 * REVISION:
 *
 */

import React					from 'react';									// eslint-disable-line no-unused-vars
import renderer					from 'react-test-renderer';
import getElementWithContext	from 'react-test-context-provider';				// for context provider
import { ThemeProvider }		from '@mui/styles';								// for custom theme
import { StyledEngineProvider, CssBaseline}	from '@mui/material';				// for jss and reset.css

import r3Theme					from '../../src/components/r3theme';			// custom theme
import R3Container				from '../../src/components/r3container';
import R3Provider				from '../../src/util/r3provider';

import mock_fetch				from '../__mocks__/fetchMock';					// eslint-disable-line no-unused-vars
import { createNodeMock }		from '../__mocks__/materialUiMock';				// for material-ui

describe('R3Container', () => {										// eslint-disable-line no-undef
	it('test snapshot for R3Container', () => {						// eslint-disable-line no-undef
		const r3provider	= new R3Provider(null);

		const element		= getElementWithContext(
			{
				r3Context:	r3provider.getR3Context()
			},
			<StyledEngineProvider injectFirst>
				<ThemeProvider theme={ r3Theme } >
					<CssBaseline />
					<R3Container
						title='K2HR3'
					/>
				</ThemeProvider>
			</StyledEngineProvider>
		);

		const component = renderer.create(element, { createNodeMock });
		let	tree = component.toJSON();
		expect(tree).toMatchSnapshot();								// eslint-disable-line no-undef
	});
});

/*
 * VIM modelines
 *
 * vim:set ts=4 fenc=utf-8:
 *
 */
