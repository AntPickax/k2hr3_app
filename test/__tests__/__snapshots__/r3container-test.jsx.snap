// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`R3Container test snapshot for R3Container 1`] = `
Array [
  <header
    className="MuiPaper-root-19 MuiPaper-elevation4-25 MuiAppBar-root-10 MuiAppBar-positionStatic-14 MuiAppBar-colorPrimary-17 R3AppBar-root-3"
  >
    <div
      className="MuiToolbar-root-46 MuiToolbar-regular-48 MuiToolbar-gutters-47 R3AppBar-toolbar-4"
    >
      <button
        aria-describedby={null}
        aria-haspopup="true"
        aria-label="main menu"
        className="MuiButtonBase-root-64 MuiIconButton-root-58 MuiIconButton-colorInherit-59"
        disabled={false}
        label="main-menu"
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onMouseOver={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex="0"
        title="Main menu"
        type="button"
      >
        <span
          className="MuiIconButton-label-63"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root-67"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
            <path
              d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
            />
          </svg>
        </span>
        <span
          className="MuiTouchRipple-root-303"
        />
      </button>
      <h6
        className="MuiTypography-root-80 MuiTypography-h6-97 MuiTypography-colorInherit-109 MuiTypography-noWrap-106 R3AppBar-title-5"
      >
        K2HR3
      </h6>
      <button
        aria-describedby={null}
        aria-haspopup="true"
        aria-label="signin menu"
        className="MuiButtonBase-root-64 MuiIconButton-root-58 MuiIconButton-colorInherit-59 R3AppBar-signinButton-8"
        disabled={false}
        label="signin-menu"
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onMouseOver={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex="0"
        title="Account"
        type="button"
      >
        <span
          className="MuiIconButton-label-63"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root-67"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 3c1.66 0 3 1.34 3 3s-1.34 3-3 3-3-1.34-3-3 1.34-3 3-3zm0 14.2c-2.5 0-4.71-1.28-6-3.22.03-1.99 4-3.08 6-3.08 1.99 0 5.97 1.09 6 3.08-1.29 1.94-3.5 3.22-6 3.22z"
            />
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
          </svg>
        </span>
        <span
          className="MuiTouchRipple-root-303"
        />
      </button>
    </div>
  </header>,
  <div>
    <div
      className="R3MainTree-dockedList-154"
    >
      <div>
        <header
          className="MuiPaper-root-19 MuiPaper-elevation0-21 MuiAppBar-root-10 MuiAppBar-positionStatic-14 MuiAppBar-colorPrimary-17 R3MainTree-subheaderAppbar-147"
        >
          <div
            className="MuiToolbar-root-46 MuiToolbar-regular-48 MuiToolbar-gutters-47 R3MainTree-subheaderToolbar-148"
          >
            <div
              className="MuiChip-root-162 MuiChip-colorPrimary-163 R3MainTree-chip-151"
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              role="button"
              tabIndex={-1}
            >
              <span
                className="MuiChip-label-181"
              >
                <span
                  className="MuiTypography-root-80 MuiTypography-subtitle2-99 R3MainTree-chipText-152"
                >
                  TENANT
                </span>
              </span>
            </div>
            <span
              className="MuiTypography-root-80 MuiTypography-subtitle2-99 MuiTypography-colorTextSecondary-113 R3MainTree-tenantListText-144"
            >
              No tenant list
            </span>
            <button
              aria-describedby={null}
              aria-label="select tenant"
              className="MuiButtonBase-root-64 MuiIconButton-root-58"
              disabled={false}
              label="select-tenant"
              onBlur={[Function]}
              onClick={[Function]}
              onContextMenu={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              onMouseOver={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex="0"
              title="Select tenant"
              type="button"
            >
              <span
                className="MuiIconButton-label-63"
              >
                <svg
                  aria-hidden="true"
                  className="MuiSvgIcon-root-67"
                  focusable="false"
                  role="presentation"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M0 0h24v24H0z"
                    fill="none"
                  />
                  <path
                    d="M12 8c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm0 2c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm0 6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
                  />
                </svg>
              </span>
              <span
                className="MuiTouchRipple-root-303"
              />
            </button>
          </div>
        </header>
      </div>
      <div
        className="MuiList-root-187"
        id="mainTree-top-list"
      >
        <div>
          <div
            className="MuiButtonBase-root-64 MuiListItem-root-191 MuiListItem-default-194 MuiListItem-gutters-199 MuiListItem-button-200"
            onBlur={[Function]}
            onClick={[Function]}
            onContextMenu={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            role="button"
            tabIndex="0"
          >
            <div
              className="MuiListItemIcon-root-203"
            >
              <svg
                aria-hidden="true"
                className="MuiSvgIcon-root-67"
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M0 0h24v24H0z"
                  fill="none"
                />
                <path
                  d="M17.63 5.84C17.27 5.33 16.67 5 16 5L5 5.01C3.9 5.01 3 5.9 3 7v10c0 1.1.9 1.99 2 1.99L16 19c.67 0 1.27-.33 1.63-.84L22 12l-4.37-6.16z"
                />
              </svg>
            </div>
            <div
              className="MuiListItemText-root-204 R3MainTree-listItemText-159"
            >
              <span
                className="MuiTypography-root-80 MuiTypography-subtitle2-99 MuiTypography-colorTextSecondary-113 MuiListItemText-primary-207"
              >
                SERVICE
              </span>
            </div>
            <span
              className="MuiTouchRipple-root-303"
            />
          </div>
        </div>
        <div>
          <div
            className="MuiButtonBase-root-64 MuiListItem-root-191 MuiListItem-default-194 MuiListItem-gutters-199 MuiListItem-button-200"
            onBlur={[Function]}
            onClick={[Function]}
            onContextMenu={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            role="button"
            tabIndex="0"
          >
            <div
              className="MuiListItemIcon-root-203"
            >
              <svg
                aria-hidden="true"
                className="MuiSvgIcon-root-67"
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M0 0h24v24H0z"
                  fill="none"
                />
                <path
                  d="M17.63 5.84C17.27 5.33 16.67 5 16 5L5 5.01C3.9 5.01 3 5.9 3 7v10c0 1.1.9 1.99 2 1.99L16 19c.67 0 1.27-.33 1.63-.84L22 12l-4.37-6.16z"
                />
              </svg>
            </div>
            <div
              className="MuiListItemText-root-204 R3MainTree-listItemText-159"
            >
              <span
                className="MuiTypography-root-80 MuiTypography-subtitle2-99 MuiTypography-colorTextSecondary-113 MuiListItemText-primary-207"
              >
                ROLE
              </span>
            </div>
            <span
              className="MuiTouchRipple-root-303"
            />
          </div>
        </div>
        <div>
          <div
            className="MuiButtonBase-root-64 MuiListItem-root-191 MuiListItem-default-194 MuiListItem-gutters-199 MuiListItem-button-200"
            onBlur={[Function]}
            onClick={[Function]}
            onContextMenu={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            role="button"
            tabIndex="0"
          >
            <div
              className="MuiListItemIcon-root-203"
            >
              <svg
                aria-hidden="true"
                className="MuiSvgIcon-root-67"
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M0 0h24v24H0z"
                  fill="none"
                />
                <path
                  d="M17.63 5.84C17.27 5.33 16.67 5 16 5L5 5.01C3.9 5.01 3 5.9 3 7v10c0 1.1.9 1.99 2 1.99L16 19c.67 0 1.27-.33 1.63-.84L22 12l-4.37-6.16z"
                />
              </svg>
            </div>
            <div
              className="MuiListItemText-root-204 R3MainTree-listItemText-159"
            >
              <span
                className="MuiTypography-root-80 MuiTypography-subtitle2-99 MuiTypography-colorTextSecondary-113 MuiListItemText-primary-207"
              >
                RESOURCE
              </span>
            </div>
            <span
              className="MuiTouchRipple-root-303"
            />
          </div>
        </div>
        <div>
          <div
            className="MuiButtonBase-root-64 MuiListItem-root-191 MuiListItem-default-194 MuiListItem-gutters-199 MuiListItem-button-200"
            onBlur={[Function]}
            onClick={[Function]}
            onContextMenu={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            role="button"
            tabIndex="0"
          >
            <div
              className="MuiListItemIcon-root-203"
            >
              <svg
                aria-hidden="true"
                className="MuiSvgIcon-root-67"
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M0 0h24v24H0z"
                  fill="none"
                />
                <path
                  d="M17.63 5.84C17.27 5.33 16.67 5 16 5L5 5.01C3.9 5.01 3 5.9 3 7v10c0 1.1.9 1.99 2 1.99L16 19c.67 0 1.27-.33 1.63-.84L22 12l-4.37-6.16z"
                />
              </svg>
            </div>
            <div
              className="MuiListItemText-root-204 R3MainTree-listItemText-159"
            >
              <span
                className="MuiTypography-root-80 MuiTypography-subtitle2-99 MuiTypography-colorTextSecondary-113 MuiListItemText-primary-207"
              >
                POLICY
              </span>
            </div>
            <span
              className="MuiTouchRipple-root-303"
            />
          </div>
        </div>
      </div>
    </div>
    <div>
      <div>
        <header
          className="MuiPaper-root-19 MuiPaper-elevation0-21 MuiAppBar-root-10 MuiAppBar-positionStatic-14 MuiAppBar-colorPrimary-17 R3Toolbar-root-210"
        >
          <div
            className="MuiToolbar-root-46 MuiToolbar-regular-48 MuiToolbar-gutters-47 R3Toolbar-toolbar-211"
          >
            <div
              aria-describedby={null}
              aria-label="current display item and its path"
              className="MuiChip-root-162 MuiChip-colorPrimary-163 MuiChip-clickable-165 MuiChip-clickableColorPrimary-166 R3Toolbar-chip-213"
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              onMouseOver={[Function]}
              onTouchEnd={[Function]}
              onTouchStart={[Function]}
              role="button"
              tabIndex={0}
              title="Display path information"
            >
              <div
                className="MuiAvatar-root-219 MuiAvatar-colorDefault-220 MuiChip-avatar-174 R3Toolbar-avatar-216 MuiChip-avatarColorPrimary-175"
              >
                <svg
                  aria-hidden="true"
                  className="MuiSvgIcon-root-67 MuiChip-avatarChildren-177 R3Toolbar-descriptionIcon-217"
                  focusable="false"
                  role="presentation"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M0 0h24v24H0z"
                    fill="none"
                  />
                  <path
                    d="M14 2H6c-1.1 0-1.99.9-1.99 2L4 20c0 1.1.89 2 1.99 2H18c1.1 0 2-.9 2-2V8l-6-6zm2 16H8v-2h8v2zm0-4H8v-2h8v2zm-3-5V3.5L18.5 9H13z"
                  />
                </svg>
              </div>
              <span
                className="MuiChip-label-181"
              >
                <span
                  className="MuiTypography-root-80 MuiTypography-subtitle2-99 R3Toolbar-chipText-214"
                >
                  Unselected
                </span>
              </span>
            </div>
            <span
              className="MuiTypography-root-80 MuiTypography-subtitle2-99 MuiTypography-colorTextSecondary-113 MuiTypography-noWrap-106 R3Toolbar-title-212"
            >
              
            </span>
            <div
              className="R3Toolbar-spacerInToolbar-218"
            />
          </div>
        </header>
      </div>
      <div
        className="MuiPaper-root-19 MuiPaper-elevation2-23 MuiPaper-rounded-20 R3MsgBox-root-273"
      >
        <svg
          aria-hidden="true"
          className="MuiSvgIcon-root-67 R3MsgBox-informationIcon-279"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M0 0h24v24H0V0z"
            fill="none"
          />
          <path
            d="M11 7h2v2h-2zM11 11h2v6h-2z"
          />
          <path
            d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8z"
          />
        </svg>
        <span
          className="MuiTypography-root-80 MuiTypography-body2-88 R3MsgBox-dialogInformationContentText-276"
        >
          First select the tenant, then select ROLE / RESOURCE / POLICY, then browse and edit.
          <br />
        </span>
      </div>
    </div>
  </div>,
]
`;
